### PR TITLE
Add prepare-for-hearing callback

### PR DESCRIPTION
**Pull request by jacmurphy**
COS: https://github.com/hmcts/finrem-case-orchestration-service/pull/303

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FR-864

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -7,13 +7,17 @@ uk.gov.notify.email.templates=${UK_GOV_NOTIFY_EMAIL_TEMPLATES:\
...
   FR_ASSIGNED_TO_JUDGE: {frEmail:'HMCTSFinancialRemedy@justice.gov.uk', contactNumber:'0300 303 0642', linkToSmartSurvey:'http://www.smartsurvey.co.uk/s/KCECE/' },\
   FR_CONSENT_ORDER_MADE: {frEmail:'HMCTSFinancialRemedy@justice.gov.uk', contactNumber:'0300 303 0642', linkToSmartSurvey:'http://www.smartsurvey.co.uk/s/KCECE/' },\
   FR_CONSENT_ORDER_NOT_APPROVED: {frEmail:'HMCTSFinancialRemedy@justice.gov.uk', contactNumber:'0300 303 0642', linkToSmartSurvey:'http://www.smartsurvey.co.uk/s/KCECE/'},\
   FR_CONSENT_ORDER_AVAILABLE: {frEmail:'HMCTSFinancialRemedy@justice.gov.uk', contactNumber:'0300 303 0642', linkToSmartSurvey:'http://www.smartsurvey.co.uk/s/KCECE/'},\
-  FR_CONTESTED_HWF_SUCCESSFUL: {frEmail:'', contactNumber:'',linkToSmartSurvey:'http://www.smartsurvey.co.uk/s/KCECE/'}}
+  FR_CONTESTED_HWF_SUCCESSFUL: {frEmail:'', contactNumber:'',linkToSmartSurvey:'http://www.smartsurvey.co.uk/s/KCECE/'},\
+  FR_CONTESTED_PREPARE_FOR_HEARING: {frEmail:'HMCTSFinancialRemedy@justice.gov.uk', contactNumber:'0300 303 0642',linkToSmartSurvey:'http://www.smartsurvey.co.uk/s/KCECE/'}\
+}
```
Comment (jacmurphy) [inline src/main/resources/application.properties:20]: May @shaunganley and I get feedback on whether this information is correct - @johnoneill46 / @yarpo

Comment (yarpo) [reply src/main/resources/application.properties:20]: I'm afraid I don't know.

Comment (yarpo) [reply src/main/resources/application.properties:20]: Where did you get it from?

Comment (jacmurphy) [reply src/main/resources/application.properties:20]: I copied the data from above in the file. But even that is inconsistent

File: src/main/java/uk/gov/hmcts/reform/finrem/notifications/controllers/NotificationController.java [hunk 2]
```diff
@@ -128,4 +129,20 @@
...
+    @ApiOperation(value = "send e-mail for 'Prepare for hearing'.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 204, message = "Prepare for hearing e-mail sent successfully")})
+    public ResponseEntity<Void> sendPrepareForHearingEmail(
+        @RequestBody
+        @ApiParam(value = "The fixtures contains case reference number, "
+            + "solicitorReferenceNumber and the email address that will receive "
+            + "the notification that the case is in the 'Prepare for a hearing' state")
+        final NotificationRequest notificationRequest) {
+        log.info("Received request to send email to Solicitor for 'Prepare for hearing' for Case ID: {}",
```
[the remark this answers is not in the document]
Comment (jacmurphy) [reply src/main/java/uk/gov/hmcts/reform/finrem/notifications/controllers/NotificationController.java:145]: Added

File: src/main/java/uk/gov/hmcts/reform/finrem/notifications/controllers/NotificationController.java [hunk 3]
```diff
@@ -128,4 +131,20 @@
...
+    @ApiResponses(value = {
+        @ApiResponse(code = 204, message = "Prepare for hearing e-mail sent successfully")})
+    public ResponseEntity<Void> sendPrepareForHearingEmail(
+        @RequestBody
+        @ApiParam(value = "The fixtures contains case reference number, "
+            + "solicitorReferenceNumber and the email address that will receive "
+            + "the notification that the case is in the 'Prepare for a hearing' state")
+        @Valid final NotificationRequest notificationRequest) {
+        log.info("Received request to send email to Solicitor for 'Prepare for hearing' for Case ID: {}",
+            notificationRequest);
```
Comment (johnoneill46) [inline src/main/java/uk/gov/hmcts/reform/finrem/notifications/controllers/NotificationController.java:146]: No need to output the entire 'notificationRequest' object, should just output the case ID by changing this to: 'notificationRequest.getCaseReferenceNumber()'

File: src/test/java/uk/gov/hmcts/reform/finrem/notifications/controllers/NotificationControllerTest.java [hunk 4]
```diff
@@ -20,11 +20,15 @@
 import uk.gov.hmcts.reform.finrem.notifications.service.EmailService;
 
 import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
```
Comment (johnoneill46) [inline src/test/java/uk/gov/hmcts/reform/finrem/notifications/controllers/NotificationControllerTest.java:24]: Couple unused imports here, be worth removing these for cleaner code

Comment (jacmurphy) [reply src/test/java/uk/gov/hmcts/reform/finrem/notifications/controllers/NotificationControllerTest.java]: Removed

File: src/main/java/uk/gov/hmcts/reform/finrem/notifications/controllers/NotificationController.java (hunk 3 above)
Comment (jacmurphy) [reply src/main/java/uk/gov/hmcts/reform/finrem/notifications/controllers/NotificationController.java]: Good spot. Thanks for the suggestion. Updated now and pushed

**Pull request merged**